R/junction-read-count.R

```diff
@@ -36,6 +48,20 @@ calculateJunctionReadCounts <- function(exonRanges, intronRanges, junctionFilePa
     seqlevelsStyle(junctionTable) <- 'UCSC'
     junctionTable$score <- junctionTable$um_reads  # to match the tophat style, uniquely mapped reads are used as score
     print('File loaded into memory')
+  } else if (junctionType == 'bam') {
+    print(paste0('Processing: ', junctionFilePath))
+    if (is.null(genome)) {
+      stop('Error: Please specify genome.')
+    }
+    rawBam <- readGAlignments(junctionFilePath)
+    bam <- keepStandardChromosomes(rawBam, pruning.mode = 'coarse')
+    rm(rawBam); gc()
```

Comment:
you can split this into 2 lines (without ";"). It's better to avoid gc(), but here it will help with the memory. We can make this function more efficient later